process/datadog_checks/process/process.py

```diff
@@ -186,11 +186,16 @@ def find_pids(self, name, search_string, exact_match, ignore_ad=True):
                             break
 
             if not matching_pids:
-                self.log.debug(
-                    "Unable to find process named %s among processes: %s",
-                    search_string,
-                    ', '.join(sorted(proc.name() for proc in self.process_list_cache.elements)),
-                )
+                # Allow debug logging while preserving warning check state.
+                # Uncaught psutil exceptions trigger an Error state
+                try:
+                    self.log.debug(
+                        "Unable to find process named %s among processes: %s",
+                        search_string,
+                        ', '.join(sorted(proc.name() for proc in self.process_list_cache.elements)),
+                    )
+                except (psutil.NoSuchProcess, psutil.AccessDenied):
+                    pass
```

Comment:
Oops — realized I had not submitted my comment here:

(Nit) Consider isolating code that could trigger the exception case, for readability purposes:

```suggestion
                # Allow debug logging while preserving warning check state.
                # Uncaught psutil exceptions trigger an Error state
                try:
                    processes = sorted(proc.name() for proc in self.process_list_cache.elements)
                except (psutil.NoSuchProcess, psutil.AccessDenied):
                    pass
                else:
                    self.log.debug(
                        "Unable to find process named %s among processes: %s", search_string, ', '.join(processes)
                    )
```

Reply:
Just fyi this log should be already happening further up in the actual processor code, just the logging code is reiterating through the list and reraising an error that was already handled. Looks like you resolved this, just providing additional context.